src/helpers/get-batched-streams.ts

```diff
@@ -5,15 +5,20 @@ import { getBatchingParams } from './get-batching-params';
 import { PagingStream } from '../paging-stream';
 import { getPagingStream } from './get-paging-stream';
 
-export const getBatchedStreams = async (request: IContentSearchRequest): Promise<PagingStream[]> => {    
-  const { numBatches, pagesPerBatch, pageSize } = await getBatchingParams(request);
-  const pageKeys: string[] = await getBatchPageKeys(numBatches, pagesPerBatch, pageSize);
+export const getBatchedStreams = async (request: IContentSearchRequest, limit?: number | undefined): Promise<PagingStream[]> => {    
+  const { numBatches, pagesPerBatch, pageSize } = await getBatchingParams(request, limit);
+  const pageKeys: string[] = await getBatchPageKeys( 
+    isNaN(limit) ? numBatches : Math.ceil(limit / (pagesPerBatch * pageSize)), 
```

Comment:
I'm a little confused by this. The first argument to getBatchPageKeys is the number of batches, which you got in line 9 from getBatchingParams. Why is further work needed here?

Reply:
Yes, if limit exists - regular batch calculation need to revised based on the calculated `pageSize` and `pagesPerBatch`. If not, our logic to omit default paging strategy in the last batch will not work.

Reply:
If thats the case, the calculation should then be done in `getBatchingParams`. Clients of that function should not have to do anything else but take those batching parameters

Reply:
@drspacemanphd I agree. I have hidden the batch revision login inside `getBatchingParams`